app/renderer/utils/locator-generation.js

```diff
@@ -70,16 +78,24 @@ export function getSimpleSuggestedLocators(attributes, sourceDoc, isNative = tru
  *
  * @param {string} path a dot-separated string of indices
  * @param {Document} sourceDoc
+ * @param {boolean} isNative whether native context is active
+ * @param {string} automationName
  * @returns {Object.<string, string>} mapping of strategies to selectors
  */
-export function getComplexSuggestedLocators(path, sourceDoc) {
+export function getComplexSuggestedLocators(path, sourceDoc, isNative, automationName) {
```

Comment:
mathods having more than 3 arguments are hard to read. Maybe we could combine some args into an object

Reply:
same below

Reply:
Among all 3 related functions (`getSuggestedLocators`, `getSimpleSuggestedLocators`, `getComplexSuggestedLocators`), there is only one common parameter (`isNative`). 4 parameters isn't _that_ bad in my opinion.

Reply:
I would say there is still a lot space for [improvements](https://www.baeldung.com/java-best-practices-many-parameters-method). Ofc, not necessarily as part of this PR.

Reply:
Oh certainly - a lot of the Inspector code is very much in need of better code quality.